autoprotocol/protocol.py

```diff
@@ -654,6 +655,117 @@ def get_instruction_index(self):
             raise ValueError("Instruction index less than 0")
         return instruction_index
 
+    @staticmethod
+    def check_informatics_field(informatics, all_wells):
+        """
+        Helper function to verify that informatics dict has a valid type
+        and data schema
+
+        Parameters
+        ----------
+        informatics: list
+            list of dict with informatics type and data
+        all_wells: WellGroup
+            All wells used in the instruction that are applicable for informatics
+            to take effect on.
+
+        Returns
+        -------
+        List
+            Validated list of dict with informatics details
+
+        Raises
```

Comment:
Right now you're raising an error when you encounter the first issue. So the user (I think?) will only get information about a single error at a time. They'll fix it, re-submit, see the next error, etc. If thats the case I would recommend not throwing an error, and continuing to inspect all elements in the array (where possible) and return the full list of errors to the user at once.

Reply:
Ah. Good point. Will update error handling :) thanks!

Reply:
Hm. Looked into this, but couple of things...
1. most of the errors are built on top of each other, i.e. if the first error fails, the rest is uncheckable/does not make sense to run, so it makes sense to raise error at that point. 
2. the rest of the implementation seems to rely on raising exceptions to propagate error, which are eventually handled and output as a json with error message. It may be possible to append multiple errors (instead of raising exception), but it's not clear to me whether it can be done cleanly without impacting how error is handled at the top...? Also, I think the error propagation and handling mechanism should be consistent across the board. This is reasonable suggestion but I think it warrants additional discussion outside the scope of this change.